xbmc/windows/GUIWindowScreensaver.cpp

```diff
@@ -42,6 +42,21 @@ void CGUIWindowScreensaver::Render()
   CGUIWindow::Render();
 }
 
+void CGUIWindowScreensaver::ClearBackground()
+{
+  /* Prevent clear as long addon runs (done byself if needed) */
+  if (m_addon)
+  {
+    if (m_renderClearCnt == 0)
+      return;
+
+    /* make five render calls with clear to prevent mix with cached display data */
```

Comment:
???

Reply:
Yeah that's really crazy, but without that, when you start some visualizations (which used previous screen data and no "glClear") it flickers to the wild (for example, some modes in Screensaver Solar Winds)

Reply:
This hack should under no circumstances go in.

I just noticed another thing: The screensavers render directly to display, right? If so, not calling glClear will not solve the problem for example on Wayland. We call `eglSwapBuffers` to display the content, but this is not a simple "front-back-buffer-switch" which it might have been in former times. Modern graphics drivers have multiple buffers ready and cycle through them as they see fit. Spec says: "The contents of ancillary buffers are always undefined after calling eglSwapBuffers. The contents of the color buffer are undefined if the value of the EGL_SWAP_BEHAVIOR attribute of surface is not EGL_BUFFER_PRESERVED." So the color buffer content is undefined when not calling `glClear` and the screensaver cannot rely on any specific content still being there.